CorePush/Google/FcmSender.cs

```diff
@@ -29,21 +29,51 @@ public FcmSender(FcmSettings settings, HttpClient http)
         /// https://firebase.google.com/docs/cloud-messaging/concept-options#notifications
         /// The SendAsync method will add/replace "to" value with deviceId
         /// </summary>
-        /// <param name="deviceId">Device token</param>
+        /// <param name="deviceId">Device token (will add `to` to the payload)</param>
         /// <param name="payload">Notification payload that will be serialized using Newtonsoft.Json package</param>
-        /// <exception cref="HttpRequestException">Throws exception when not successful</exception>
-        public async Task<FcmResponse> SendAsync(string deviceId, object payload)
+        /// <cref="HttpRequestException">Throws exception when not successful</exception>
+        public Task<FcmResponse> SendAsync(string deviceId, object payload)
         {
             var jsonObject = JObject.FromObject(payload);
             jsonObject.Remove("to");
             jsonObject.Add("to", JToken.FromObject(deviceId));
-            var json = jsonObject.ToString();
 
+            return SendRawAsync(jsonObject.ToString());
+        }
+
+        /// <summary>
+        /// Send firebase notification.
+        /// Please check out payload formats:
+        /// https://firebase.google.com/docs/cloud-messaging/concept-options#notifications
+        /// The SendAsync method will add/replace "to" value with deviceId
+        /// </summary>
+        /// <param name="payload">Notification payload that will be serialized using Newtonsoft.Json package</param>
+        /// <exception cref="HttpRequestException">Throws exception when not successful</exception>
+        public Task<FcmResponse> SendAsync(object payload)
+        {
+            return SendRawAsync(JsonHelper.Serialize(payload));
+        }
+
+        /// <summary>
+        /// Send firebase notification.
+        /// Please check out payload formats:
+        /// https://firebase.google.com/docs/cloud-messaging/concept-options#notifications
+        /// The SendAsync method will add/replace "to" value with deviceId
+        /// </summary>
+        /// <param name="payload">Notification payload json</param>
+        /// <exception cref="HttpRequestException">Throws exception when not successful</exception>
+        public async Task<FcmResponse> SendRawAsync(string payload)
```

Comment:
@danielgindi can you fix a couple more things:

1. Let's not have this as public for now. I want to make sure we don't open too many different options for users. Lack of them sometimes makes things simpler. :)
2. Can you add `SendAsync(object payload)` to IFcmSender interface.

Thank you!

Reply:
@andrei-m-code Done :-)
Although I see no reason to prevent users from encoding their own payloads